pkg/utils/k8sutil.go

```diff
@@ -0,0 +1,117 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/keikoproj/addon-manager/pkg/client/informers/externalversions/internalinterfaces"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/dynamic"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
+)
+
+// GetClient returns a k8s clientset to the request from inside of cluster
+func GetClient() kubernetes.Interface {
```

Comment:
Why would we do this, this is reimplementing controller-runtime?

```
kubernetes.NewForConfigOrDie(ctrl.GetConfigOrDie())
```

Reply:
This is a generic way creating a k8s client - leveraging kubernetes.NewForConfig(config) with in-cluster configuration.

kubernetes.NewForConfigOrDie(cfg *rest.config), you are lacking a way retrieving kubeconfig